Fix up build-wheels script

janden opened the pull request:
This fixes a couple of issues:
- It runs `make clean` to remove any remnants of already built libraries.
- It adds a `make.inc.manylinux` for greater compatibility with older chips (non-AVX).
- It updates and adds some comments.

Somehow I had written these a while ago and they never got into a PR.

ahbarnett reviewed (approved):
I am leaving the CI & wheels up to Libin & Joakim & Garrett.
So with caveat that I don't know what your status is here, I approve.

Minor suggestion: the file contains:
Copyright (c) 2019, Henry Schreiner
It is supposed to come with a LICENSE file, but does not. That BSD3 license differs from our Apache2, so limits a little what FINUFFT can be done with.
Is this file short enough, and we changed it enough, that (C) no longer applies?

janden commented:
> Is this file short enough, and we changed it enough, that (C) no longer applies?

I'm not sure. @lu1and10 do you know what the original looked like?

lu1and10 commented:
> > Is this file short enough, and we changed it enough, that (C) no longer applies?
> 
> I'm not sure. @lu1and10 do you know what the original looked like?

Now, the common lines with the original file are lines 40-47 which, we have changed all the other lines. I guess that (C) no longer applies?

janden commented:
> Now, the common lines with the original file are lines 40-47 which, we have changed all the other lines. I guess that (C) no longer applies?

I think so. Removed them.
